src/bootstrap/compile.rs

```diff
@@ -1295,6 +1295,14 @@ impl Step for Assemble {
         // when not performing a full bootstrap).
         builder.ensure(Rustc::new(build_compiler, target_compiler.host));
 
+        // Do early return instead of assembling for stage3(since it has problems
+        // with sysroot) if full-bootstrap isn't used.
+        //
+        // See: https://github.com/rust-lang/rust/issues/90244
+        if target_compiler.stage >= 3 && !builder.config.full_bootstrap {
```

Comment:
I'm not convinced this isn't just hiding the problem, it looks like we would presumably still run into the same problem with full-bootstrap and stage 3? Or, if not, then the fix is probably in some other conditional on stage 3 / full bootstrap elsewhere, which should treat stage 2 similarly.

I'm OK merging this since full-bootstrap use is rarer and it sounds like the issue is worth at least patching over, but I would prefer that this comment indicate that this is a hack. Something like:

> FIXME: For now patch over problems noted in #90244 by early returning here, even though we've not properly assembled the target sysroot. A full fix is pending further investigation, for now full bootstrap usage is rare enough that this is OK.

Reply:
> I'm not convinced this isn't just hiding the problem, it looks like we would presumably still run into the same problem with full-bootstrap and stage 3?

Yes, this fixes the problem where we only want to build stage 2 compiler and get stage 3 error. Doesn't fix the main problem with stage3.

Reply:
Hm. So I re-ran with full-bootstrap and stage=3 and stage=4 and that seems to work... I still suspect this isn't quite right (feels like the wrong place for this to live) but for now let's just accept as-is.

Reply:
I haven't done detailed debug on the stage3 problem. I was focused on the hotfixing stage3 error on stage2 compilation. I could check that one too if you want.